framework/src/interfaces/Coupleable.C

```diff
@@ -492,6 +498,60 @@ Coupleable::coupledVectorTagValue(const std::string & var_name, TagID tag, unsig
     return var->vectorTagValue(tag);
 }
 
+const VariableValue &
+Coupleable::coupledVectorTagValue(const std::string & var_name,
+                                  const std::string & tag_name,
+                                  unsigned int comp) const
+{
+  if (!_c_parameters.isParamValid(tag_name))
+    return coupledValue(var_name, comp);
```

Comment:
Typically the parameter of tag name if for a particular iterative solver and the same kernel is also for our general solver which does not require the tag name parameter. We could move this out side of this function in kernel's constructor header, but my argument is that this will make the normal use case tedious. If a user want, they can always do this check on their own. `CoupledForceLagged` gives an example. This is kind of equivalent with having the default tag name for the current solution vector. If we have that tag, users can put that tag name in the parameter as the default value. If the tag name parameter is required, like in `TagVectorAux`, we will not hit this early return. Does this convince you?

Reply:
My concern is "return coupledValue(var_name, comp);" when users do not set tag name or the tag name does not exist.

In other words, I want to separate this API from `coupledValue`. I do not want this API call `coupledValue`. 

When users call this API and also want to use `coupledValue` when tag name does not exist,  users should do their `if ` logic .

Reply:
What I want to is 

```
if (!_c_parameters.isParamValid(tag_name) )
  MooseError("You have to specify tag name if you want to use  coupledVectorTagValue ")
  

  TagName tagname = _c_parameters.get<TagName>(tag_name);
  if (tagname.empty())
    MooseError("You can not use coupledVectorTagValue because tag name does not exist ")
```

Reply:
I updated this part as you suggested. The only difference is that I added a new tag name for users to indicate that they want to use the same interface sometimes for coupling the current solution.